app/src/main/java/com/pajato/argus/ListAdapter.kt

```diff
@@ -14,20 +15,38 @@ class ListAdapter(val items: MutableList<Video>) : RecyclerView.Adapter<ListAdap
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        val titleTextView: TextView = holder.layout.findViewWithTag("title")
+        // Fill in the data and set an onClick for the delete button
+        val titleTextView = holder.layout.findViewWithTag<TextView>("title")
         titleTextView.text = items[position].title
-        val networkTextView: TextView = holder.layout.findViewWithTag("network")
+        val networkTextView = holder.layout.findViewWithTag<TextView>("network")
         networkTextView.text = items[position].network
+        val deleteButton = holder.layout.findViewWithTag<AppCompatImageButton>("delete")
+        deleteButton.setOnClickListener(Delete(holder, this))
     }
 
     override fun getItemCount(): Int {
         return items.size
     }
 
-    fun addItem(video : Video) {
+    fun addItem(video: Video) {
         items.add(video)
         this.notifyItemInserted(items.size - 1)
     }
 
+    fun removeItem(position : Int) {
+        items.removeAt(position)
+        notifyItemRemoved(position)
+        notifyItemRangeChanged(position, items.size)
+    }
+
+    // The ViewHolder class
     class ViewHolder(val layout: View) : RecyclerView.ViewHolder(layout)
+
+    // An onClick that deletes the item from the adapter and removes it from the database.
+    class Delete(private val holder: ViewHolder, private val adapter: ListAdapter) : View.OnClickListener {
```

Comment:
I'm getting a sense that you are discovering the need for RxJava (aka event bus) even though you may not yet realize it.

Reply:
I was thinking about this earlier today, and while I don't think we need RxJava everywhere -- or maybe not even anywhere yet -- it would make the delete operation much more palatable.